sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/synthetic/_model_tools/_async_http_client.py

```diff
@@ -0,0 +1,96 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import logging
+
+from aiohttp import TraceConfig
+from aiohttp_retry import RandomRetry, RetryClient
+
+
+class AsyncHTTPClientWithRetry:
+    """Async HTTP client with retry configuration and request logging.
+
+    :param n_retry: Number of retries to attempt
+    :type n_retry: int
+    :param retry_timeout: Timeout between retries, in seconds
+    :type retry_timeout: int
+    :param logger: The logger object to use for request logging
+    :type logger: logging.Logger
+    :param retry_options: The retry options. Defaults to None.
+    :type retry_options: Optional[aiohttp_retry.retry_options.BaseRandomRetry]
+    """
+    def __init__(self, n_retry: int, retry_timeout: int, logger: logging.Logger, retry_options=None):
+        self.attempts = n_retry
+        self.logger = logger
+
+        # Set up async HTTP client with retry
+
+        trace_config = TraceConfig()  # set up request logging
+        trace_config.on_request_end.append(self.delete_auth_header)
+        if retry_options is None:
+            retry_options = RandomRetry(  # set up retry configuration
```

Comment:
All retry configuration options should be consistent with the rest of the Azure SDK experience.

Reply:
Same of course with logging configuration, and other aspects that would typically be handled in the Azure core pipeline.

Reply:
Will open a task in the backlog to do a review on consistent HTTP option handling.